### DISCUSSION TITLE
error help: Lesson 7 FundMe test > testWithdrawFromMultipleFunders keeps failing

### DISCUSSION THREAD

**Discussion by Josetoshi** [Cyfrin/foundry-full-course-cu#1252] [General]
can't seem to figure out what the problem is here. chat gpt says its hoax but code looks exactly as Patrick is guiding. 

here is the error:

'''
Running 1 test for test/FundMeTest.t.sol:FundMeTest
[FAIL. Reason: cannot overwrite a prank until it is applied at least once] testWithdrawFromMultipleFunders() (gas: 6096)
Traces:
  [6096] FundMeTest::testWithdrawFromMultipleFunders()
    ├─ [0] VM::prank(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   └─ ← ()
    ├─ [0] VM::deal(0x0000000000000000000000000000000000000001, 100000000000000000 [1e17])
    │   └─ ← ()
    ├─ [0] VM::prank(0x0000000000000000000000000000000000000001)
    │   └─ ← cannot overwrite a prank until it is applied at least once
    └─ ← cannot overwrite a prank until it is applied at least once

Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 1.91ms
 
Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/FundMeTest.t.sol:FundMeTest
[FAIL. Reason: cannot overwrite a prank until it is applied at least once] testWithdrawFromMultipleFunders() (gas: 6096)
'''

heres the full code

'''solidity

    function testWithdrawFromMultipleFunders() public funded {
        // Arrange
        uint160 numberOfFunders = 10;
        uint160 startingFunderIndex = 1;
        for (uint160 i = startingFunderIndex; i < numberOfFunders; i++){
            // vm prank new address
            // vm deal new address
            hoax(address(i), SEND_VALUE);
            fundMe.fund{value: SEND_VALUE}();
            // fund the fundMe
        }

        uint256 startingOwnerBalance = fundMe.getOwner().balance;
        uint256 startingFundMeBalance = address(fundMe).balance;
        
        // Act
        vm.startPrank(fundMe.getOwner());
        fundMe.withdraw();
        vm.stopPrank();

        // Assert
        assert(address(fundMe).balance == 0);
        assert(
            startingFundMeBalance + startingOwnerBalance == 
                fundMe.getOwner().balance
            );

    }
'''

**Comment by Josetoshi** [comment]
@PatrickAlphaC

**Reply by PatrickAlphaC** [reply]
Could you please remember to follow this section to format your questions? https://youtu.be/umepbfKp5rI?t=22464

And it looks like you have a double prank somewhere in `testWithdrawFromMultipleFunders`

**Reply by Josetoshi** [reply]
I thought i did my bad, and the only pranks im using in testWithdrawFromMultipleFunders is hoax on the arrange section and start stop prank on the act section of the function. Yours looks exactly the same on the video and it passed the test. ?

```solidity

function testWithdrawFromMultipleFunders() public funded {
        // Arrange
        uint160 numberOfFunders = 10;
        uint160 startingFunderIndex = 1;
        for (uint160 i = startingFunderIndex; i < numberOfFunders; i++) {
            // vm prank new address
            // vm deal new address
            hoax(address(i), SEND_VALUE);
            fundMe.fund{value: SEND_VALUE}();
            // fund the fundMe
        }

        uint256 startingOwnerBalance = fundMe.getOwner().balance;
        uint256 startingFundMeBalance = address(fundMe).balance;

        // Act
        vm.startPrank(fundMe.getOwner());
        fundMe.withdraw();
        vm.stopPrank();

        // Assert
        assert(address(fundMe).balance == 0);
        assert(
            startingFundMeBalance + startingOwnerBalance ==
                fundMe.getOwner().balance
        );
    }
```

**Reply by PatrickAlphaC** [reply]
Close! You did 3 quotes (''') it needs to be 3 backticks (```) I fixed it for you above, see the difference? 

And yes, could you try @Abbasjafri-syed's suggestion? Still looks like you have 2 pranks somewhere.

**Comment by Josetoshi** [comment]
still stuck here. I've tried everything all the ai sites and still don't undertand how to fix with my limited knowledge. Should i just skip this test and continue with the lesson? please help

**Reply by Abbasjafri-syed** [reply]
try 
```
// Act
    vm.prank(fundMe.getOwner());
    fundMe.withdraw();
```
instead of 
```
// Act
    vm.startPrank(fundMe.getOwner());
    fundMe.withdraw();
    vm.stopPrank();
```

**Comment by Josetoshi** [comment]
Appreciate the input but it didn't work. Instead of hoax I changed it to deal 
`deal(address(i), SEND_VALUE);
            assertEq(address(i).balance, SEND_VALUE);`

and it seems to have worked! I'm moving on lol

**Reply by Abbasjafri-syed** [reply]
yes identified the issue as hoax is combination of both prank and deal...
your code here 
```
hoax(address(i), SEND_VALUE);
```
eliminate the need of ```    vm.startPrank(fundMe.getOwner());```

**Reply by ghost** [reply]
Hello, I was having the same issue. This fix worked for me as well. Thank you

**Reply by EngrPips** [reply]
Interesting
